src/AspNet.Security.OAuth.DingTalk/DingTalkAuthenticationDefaults.cs

```diff
@@ -0,0 +1,53 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+
+namespace AspNet.Security.OAuth.DingTalk
+{
+    /// <summary>
+    /// Default values for DingTalk authentication.
+    /// </summary>
+    public static class DingTalkAuthenticationDefaults
+    {
+        /// <summary>
+        /// Default value for <see cref="AuthenticationScheme.Name"/>.
+        /// </summary>
+        public const string AuthenticationScheme = "DingTalk";
+
+        /// <summary>
+        /// Default value for <see cref="AuthenticationScheme.DisplayName"/>.
+        /// </summary>
+        public const string DisplayName = "DingTalk";
+
+        /// <summary>
+        /// Default value for <see cref="AuthenticationSchemeOptions.ClaimsIssuer"/>.
+        /// </summary>
+        public const string Issuer = "DingTalk";
+
+        /// <summary>
+        /// Default value for <see cref="RemoteAuthenticationOptions.CallbackPath"/>.
+        /// </summary>
+        public const string CallbackPath = "/signin-dingtalk";
+
+        /// <summary>
+        /// Default value for <see cref="OAuthOptions.AuthorizationEndpoint"/>.
+        /// </summary>
+        public const string AuthorizationEndpoint = "https://oapi.dingtalk.com/connect/qrconnect";
```

Comment:
Why use a QR code over the more usual "credentials" approach, which is what the other URL seems to be pointing to?

Reply:
Because the login methods include password login and app scan code login, scan code login is the main habit

因为 登入方式分为密码登入及app扫码登入，以扫码登入为主要习惯

Reply:
If there's a "recommended" approach, we should just support that one. If users want to use the other one, they can update the configuration themselves. Otherwise we have more things to maintain.

Reply:
`DingTalkAuthenticationOptions.IsUseUserPasswordLogin = true` can be configured to use password mode to log in. The default is scan code login

Reply:
I get that, but do we really need the extra code and property, when a user could just set this?

```csharp
options.AuthorizationEndpoint = "https://oapi.dingtalk.com/connect/oauth2/sns_authorize";
```

I also notice that there's a scope with SNS in the name, as well as SNS in the path of the token and user endpoints, where as the QR code one doesn't. The username-password URL also has OAuth in the path, where as the QR code one doesn't.

Is the DingTalk API inconsistent, or is this PR mixing different concepts together that it shouldn't be?

Reply:
Yes, QR code is in the document (https://ding-doc.dingtalk.com/docා/serverapi3/mrugr3). The username password URL is in a different document (https://ding-doc.dingtalk.com/docා/serverapi3/sz4hm2). However, I find that they are compatible with each other and obtain appid and appsecret in the same place